### PR TITLE
Add ability to preview modular pages

### PR DESCRIPTION
**Pull request by k7hpn**

### CODE REVIEW

File: src/Promenade.Models/Entities/PageLayout.cs [hunk 1]
```diff
@@ -30,5 +30,10 @@ public class PageLayout
 
         [NotMapped]
         public PageLayoutText PageLayoutText { get; set; }
+
+        public Guid PreviewId { get; set; }
+
+        [NotMapped]
+        public bool IsPreview { get; set; }
```
Comment (iafb) [inline src/Promenade.Models/Entities/PageLayout.cs:37]: I think this field can be removed, it doesn't like its value is being used anywhere.

File: src/Promenade.Web/Views/Shared/LayoutPage.cshtml [hunk 2]
```diff
@@ -33,6 +33,18 @@
     </div>
 }
 
+@if (Model.PageLayout.IsPreview)
+{
+    <div class="row">
+        <div class="col-12 col-md-6 offset-md-3">
+            <div class="alert alert-info">
+                You are viewing a preview of how this page will appear as of
+                @Model.PageLayout.StartDate.
```
Comment (iafb) [inline src/Promenade.Web/Views/Shared/LayoutPage.cshtml:42]: This message could be a bit confusing of there's no start date set for the layout.

File: src/Promenade.Controllers/NewsController.cs [hunk 3]
```diff
@@ -24,13 +24,20 @@ protected override PageType PageType
...
             return await ReturnPageAsync(stub);
         }
 
-        [Route("{stub?}/item/{id}")]
+        [HttpPost("{stub?}")]
+        public async Task<IActionResult> PagePreview(string stub)
+        {
+            return await ReturnPreviewPageAsync(stub,
+                HttpContext.Request.Query["PreviewId"]);
+        }
```
Comment (iafb) [inline src/Promenade.Controllers/NewsController.cs:38]: This should be added to the HomeController as well, I think layout pages should be able to work with the home page.

**Review by iafb — CHANGES_REQUESTED** [review]

File: src/Promenade.Models/Entities/PageLayout.cs (hunk 1 above)
Comment (k7hpn) [reply src/Promenade.Models/Entities/PageLayout.cs:37]: I think it's used in PageService.cs and LayoutPage.cshtml.

File: src/Promenade.Controllers/NewsController.cs (hunk 3 above)
Comment (k7hpn) [reply src/Promenade.Controllers/NewsController.cs:38]: Agreed. Was thinking we'd do this later but now seems like the right time.

File: src/Promenade.Web/Views/Shared/LayoutPage.cshtml (hunk 2 above)
Comment (k7hpn) [reply src/Promenade.Web/Views/Shared/LayoutPage.cshtml:42]: You are right, will edit to better handle a null start date. I forgot start dates could be null!

**Review by iafb — APPROVED** [review]

**Pull request merged**